sql/log_event.cc

```diff
@@ -9088,14 +9090,11 @@ int Rows_log_event::open_record_scan() {
 
   if (m_key_index < MAX_KEY) {
     if (m_rows_lookup_algorithm == ROW_LOOKUP_HASH_SCAN) {
-      /* initialize the iterator over the list of distinct keys that we have */
-      m_itr = m_distinct_keys.begin();
```

Comment:
Shouldn't we change this to `m_distinct_key_id = 0`?

Reply:
m_distinct_key_idx is already set to 0 when it is declared.

Reply:
@VarunNagaraju my point here is that you do not have this initial initialization logic in the constructor of the `Rows_log_event` class but instead have a special function `open_record_scan()` in the interface of this class.
Which means that theoretically the user of this class is allowed to do
```
open_record_scan();
while(...) {
  next_record_scan();
  ...  
}
// and then start iteration for the second time
open_record_scan();
while(...) {
  next_record_scan();
  ...  
}
```
I understand that currently this is not happening - just want to be "future changes"-proof 🙂 here.
In other words, I would still put that `m_distinct_key_id = 0` here.